tsdb/engine/tsm1/engine.go

```diff
@@ -475,6 +477,10 @@ func (e *Engine) WithLogger(log zap.Logger) {
 
 // LoadMetadataIndex loads the shard metadata into memory.
 func (e *Engine) LoadMetadataIndex(shardID uint64, index tsdb.Index) error {
+	if index.Type() != inmem.IndexName {
```

Comment:
That would be better but the `LoadMetadataIndex()` is fairly complicated between walking the file store and cache. If a refactor isn't too bad then I agree that we should push it down into the index.